contracts/AxelarGateway.sol

```diff
@@ -50,13 +50,21 @@ contract AxelarGateway is IAxelarGateway, AdminMultisigBase {
     // solhint-disable-next-line var-name-mixedcase
     address internal immutable AUTH_MODULE;
     // solhint-disable-next-line var-name-mixedcase
+    address internal immutable GOVERNANCE;
+    // solhint-disable-next-line var-name-mixedcase
     address internal immutable TOKEN_DEPLOYER_IMPLEMENTATION;
 
-    constructor(address authModule_, address tokenDeployerImplementation_) {
+    constructor(
+        address authModule_,
+        address governance_,
```

Comment:
Circular dependency issue with InterchainGovernance constructor may not be ideal

Reply:
Resolved by moving GOVERNANCE into a setup param